repository/src/main/java/tech/ydb/yoj/repository/db/cache/FirstLevelCache.java

```diff
@@ -90,60 +222,110 @@ private <E extends Entity<E>> EntityCache<E> getEntityCache() {
             }
 
             @Override
-            public <E extends Entity<E>> Optional<E> peek(@NonNull Entity.Id<E> id) {
+            public <E extends Entity<E>> Optional<E> peek(@NonNull TableDescriptor<E> tableDescriptor, @NonNull Entity.Id<E> id) {
                 EntityCache<E> cache = getEntityCache();
-                if (cache.containsKey(id)) {
-                    return cache.get(id);
+                EntityCache.Key<E> key = new EntityCache.Key<>(tableDescriptor, id);
+                if (cache.containsKey(key)) {
+                    return cache.get(key);
                 }
                 throw new NoSuchElementException();
             }
 
             @Override
-            public <E extends Entity<E>> E get(@NonNull Entity.Id<E> id, @NonNull Function<Entity.Id<E>, E> loader) {
+            public <E extends Entity<E>> E get(@NonNull TableDescriptor<E> tableDescriptor, @NonNull Entity.Id<E> id, @NonNull Function<Entity.Id<E>, E> loader) {
                 EntityCache<E> cache = getEntityCache();
-                if (cache.containsKey(id)) {
-                    return cache.get(id).orElse(null);
+                EntityCache.Key<E> key = new EntityCache.Key<>(tableDescriptor, id);
+                if (cache.containsKey(key)) {
+                    return cache.get(key).orElse(null);
                 }
 
                 E entity = loader.apply(id);
-                cache.put(id, Optional.ofNullable(entity));
+                cache.put(key, Optional.ofNullable(entity));
 
                 return entity;
             }
 
             @Override
-            public <E extends Entity<E>> List<E> snapshot(@NonNull Class<E> entityType) {
+            public <E extends Entity<E>> List<E> snapshot(@NonNull TableDescriptor<E> tableDescriptor) {
                 EntityCache<E> cache = getEntityCache();
-                return cache.values().stream()
-                        .flatMap(Optional::stream)
-                        .filter(v -> entityType.equals(v.getId().getType()))
-                        .collect(Collectors.toUnmodifiableList());
+                return cache.entrySet().stream()
+                        .filter(e -> tableDescriptor.equals(e.getKey().tableDescriptor()))
+                        .flatMap(e -> e.getValue().stream())
+                        .toList();
             }
 
             @Override
-            public <E extends Entity<E>> Map<Entity.Id<E>, E> entities(@NonNull Class<E> entityType) {
-                EntityCache<E> cache = getEntityCache();
-                return Collections.unmodifiableMap(Maps.transformValues(
-                        Maps.filterEntries(cache, e -> entityType.equals(e.getKey().getType()) && e.getValue().isPresent()),
-                        Optional::get
-                ));
+            public <E extends Entity<E>> Map<Entity.Id<E>, E> entities(@NonNull TableDescriptor<E> tableDescriptor) {
```

Comment:
Wanting to have a "live view" of the (part of) the first-level cache is *a bit strange*, I would say, but I remember that there were several uses for this in the Cloud Java codebase. @lavrukov Please check whether anybody at all uses the `FirstLevelCache.entities()` instead of the `FirstLevelCache.snapshot()`. Do they really need the "live view"? It will be a very inefficient `Map` implementation, with only `Map.get()` and `Map.containsKey()` being reasonably fast (`Map.size()` would take linear time in *the size of the whole first-level cache, for example*... because it needs to filter out irrelevant cache entries).

Reply:
Noone uses FirstLevelCache.entities(), I think it's better to remove it